MapboxVision/VisionManager/VisionReplayManager.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  VisionReplayManager.swift
+//  MapboxVision
+//
+//  Created by Alexander Pristavko on 5/17/19.
+//  Copyright © 2019 Mapbox. All rights reserved.
+//
+
+import Foundation
+import CoreMedia
+
+public final class VisionReplayManager: BaseVisionManager {
+
+    public static func create(recordPath: String) throws -> VisionReplayManager {
+        return VisionReplayManager(dependencies: try ReplayDependencies.default(recordPath: recordPath))
+    }
+
+    public var videoSource: VideoSource {
+        return dependencies.player
+    }
+
+    /**
+     Start delivering events from `VisionManager`.
+
+     - Parameter delegate: Delegate for `VisionManager`. Delegate is held as a strong reference until `stop` is called.
+     */
+    public func start(delegate: VisionManagerDelegate? = nil) {
+        switch state {
+        case .uninitialized:
+            assertionFailure("VisionManager should be initialized before starting")
+            return
+        case .started:
+            assertionFailure("VisionManager is already started")
+            return
+        case .initialized, .stopped:
+            self.delegate = delegate
+            state = .started
+        }
+
+        resume()
+    }
+
+    /**
+     Stop delivering events from `VisionManager`.
+     */
+    public func stop() {
+        guard state == .started else {
+            assertionFailure("VisionManager is not started")
+            return
+        }
+
+        pause()
+
+        state = .stopped
+        delegate = nil
+    }
+
+    /**
+     Cleanup the state and resources of `VisionManger`.
+     */
+    public func destroy() {
+        guard !state.isUninitialized else { return }
+
+        if case .started = state {
+            stop()
+        }
+
+        dependencies.native.destroy()
+        state = .uninitialized
+    }
+
+    private let dependencies: ReplayDependencies
```

Comment:
Should we move it below `// MARK: Private`?

Reply:
Ok